apply/run.go

```diff
@@ -40,22 +41,27 @@ func IsIPList(args string) bool {
 }
 
 func (c *ClusterArgs) SetClusterArgs() error {
-	if IsNumber(c.masterArgs) && IsNumber(c.nodeArgs) {
-		c.cluster.Spec.Masters.Count = c.masterArgs
-		c.cluster.Spec.Nodes.Count = c.nodeArgs
-		c.cluster.Spec.SSH.Passwd = c.passwd
-		c.cluster.Spec.Provider = common.AliCloud
-		return nil
+	c.cluster.Spec.Image = c.imageName
+	if IsNumber(c.masterArgs) {
+		if IsNumber(c.nodeArgs) || c.nodeArgs == "" {
+			c.cluster.Spec.Masters.Count = c.masterArgs
+			c.cluster.Spec.Nodes.Count = c.nodeArgs
+			c.cluster.Spec.SSH.Passwd = c.passwd
+			c.cluster.Spec.Provider = common.AliCloud
```

Comment:
c.cluster.Spec.Provider need to read from Cluster.  or set a default CloudProvider
If user want to Use other Cloud . we may suport for multi CloudProvider soon?  

```
DefaultCloudProvider = AliCloud
common.DefaultCloudProvider
```

Reply:
or add cloud provider , if masterArgs is number then replace the provider